src/stdio_mgr/types.py

```diff
@@ -0,0 +1,175 @@
+r"""``stdio_mgr.types`` *code module*.
+
+``stdio_mgr.types`` provides context managers for convenient
+interaction with ``stdin``/``stdout``/``stderr`` as a tuple.
+
+**Author**
+    John Vandenberg (jayvdb@gmail.com)
+
+**File Created**
+    6 Sep 2019
+
+**Copyright**
+    \(c) Brian Skinn 2018-2019
+
+**Source Repository**
+    http://www.github.com/bskinn/stdio-mgr
+
+**Documentation**
+    See README.rst at the GitHub repository
+
+**License**
+    The MIT License; see |license_txt|_ for full license terms
+
+**Members**
+
+"""
+import collections.abc
+from contextlib import suppress
+from io import TextIOBase
+
+from stdio_mgr.compat import AbstractContextManager
+
+
+class MultiItemTuple(collections.abc.Iterable):
+    """Iterable with methods that operate on all items."""
+
+    @staticmethod
+    def _invoke_name(item, method_name):
+        item = getattr(item, method_name)
+        if callable(item):
+            item = item()
+        return item
+
+    def _map_method(self, method_name):
```

Comment:
I'm not immediately seeing the logic for which of these are private vs public. map/all/any at least I could see changing to not collide with builtins, but then why not append the `_`, instead of prepend?

Reply:
Hmm , prefix `_` are because of lint errors if I used method names like builtins, which is a sensible lint error most of the time.  I would be happy to `_` prefix them all.

Reply:
IIRC, the standard way to disambiguate identifiers away from builtins is to append the underscore: `map_`.

These all look pretty useful to outside users; I don't think we want to mark them private.